ext/sqlite3/statement.c

```diff
@@ -162,13 +162,15 @@ step(VALUE self)
                         if (internal_encoding) {
                             val = rb_str_export_to_enc(val, internal_encoding);
                         }
+                        rb_obj_freeze(val);
```

Comment:
Hmm, this seems to break Rails's schema cache.

```
bin/test test/cases/connection_adapters/schema_cache_test.rb -n/test_gzip_dumps_identical/
```

Without this line, the test passes. With it, the test fails (in isolation).

I'm looking into it.

Reply:
Context on that test: https://github.com/rails/rails/pull/49166

Reply:
See https://github.com/rails/rails/pull/51581 for the exciting conclusion.